crates/runtime/Cargo.toml

```diff
@@ -76,8 +76,8 @@ keyring = { version = "3.6.2", features = [
 ], optional = true }
 llms = { path = "../llms" }
 logos = "0.15.0"
-mcp-client = { git = "https://github.com/block/goose", package = "mcp-client", optional = true }
-mcp-core = { git = "https://github.com/block/goose", package = "mcp-core", optional = true }
+mcp-client = { git = "https://github.com/Jeadie/mcp-rust-sdk.git", rev = "060cdd2d9e23d129894160e26c66cc82b3fe37eb", package = "mcp-client", optional = true }
+mcp-core = { git = "https://github.com/Jeadie/mcp-rust-sdk.git", rev = "060cdd2d9e23d129894160e26c66cc82b3fe37eb", package = "mcp-core", optional = true }
```

Comment:
Make a PR into https://github.com/spiceai/modelcontextprotocol-rust-sdk/tree/spiceai and raise an issue on yourself to track merging the upstream change and reverting back to the upstream crate.